docs/source-key.md

```diff
@@ -0,0 +1,127 @@
+
+#### Ambiguous
+- `question_feedbacks` is written by portal-report, but the documents are related to the answer created by the runtime (either activity player or LARA), the question created by LARA, and the assignment in the portal. The related resources can all be discovered from the answer, so we could say the source key should be the domain of the runtime which generated the answers. Currently the sourceKey is `authoring.concord.org`, which is correct for the LARA runtime but incorrect for the activity player runtime.
+- `activity_feedbacks` is written by portal-report, but the documents are related to the student's work on the assignment. Currently this associating is stored using the portal's contextId for the assignment and portalStudentId. The actual work is really from the runtime, but there not yet any document saved by the runtime representing the student's work. When the activity player saves the student's current page, then there will be a document for this work. Assuming we move towards this document of student work, then the related resource is the domain of the runtime which generated the work. Currently the sourceKey is `authoring.concord.org`, which is correct for the LARA runtime but incorrect for the activity player runtime.
+
+#### Incorrect
+- `feedback_settings` is written by portal-report, but the settings are specific to an assignment in the portal. The sourceKey is `authoring.concord.org`. This is incorrect with this design goal, it should be `learn.concord.org`.
+
+## Possible Improvement: Replace sourceKey with rootKey
+Here is an idea of way to improve the ambiguous use of sourceKeys above.
+
+A single rootKey would be shared by AP, portal-report, LARA, and researcher report.
+The rootKey would default to something like 'production', but could be overridden for staging
+and development testing.
+
+All data would have to change. AP, portal-report, LARA, and researcher report would need to support this new approach, and ideally also be backward compatible with old approach so we can do the migration easily.
+
+### What about the access rules?
+The source is not currently checked by any of the access rules.
+
+### What about conflicting documents?  
+If we don't use the source then the activity-player and LARA answer documents will be mixed in the same folder.
+
+The answer documents currently include a `tool_id`. In the AP this domain plus the path of the url of the running the activity player so it also includes the version or branch in. LARA always adds this to the answers, and it typically should be `https://authoring.concord.org` or a developer might set it to something else for testing purposes.
```

Comment:
I didn't follow the second sentence: `In the AP this domain plus the path of the url of the running the activity player so it also includes the version or branch in.`

Reply:
I updated this section to hopefully make it more clear.